mesh/amqp.c

```diff
@@ -609,16 +632,12 @@ struct mesh_amqp *mesh_amqp_new(void)
 	return amqp;
 }
 
-void mesh_amqp_start(struct mesh_amqp *amqp, struct mesh_amqp_config *config)
+void mesh_amqp_start(struct mesh_amqp *amqp)
 {
 	pthread_attr_t attr;
 	int fds[2];
-	struct amqp_thread_context *thread_context;
 
-	struct mesh_amqp_config empty_config = {0};
-	config = config ?: &empty_config;
-
-	thread_context = l_new(struct amqp_thread_context, 1);
+	struct amqp_thread_context *thread_context = l_new(struct amqp_thread_context, 1);
```

Comment:
This leaks in line 652 (early return on failure).

Reply:
fixed